### DISCUSSION TITLE
Can not install the new version - error messages in body

### DISCUSSION THREAD

**Discussion by catclaw** [lllyasviel/Fooocus#3416] [Q&A] [answered]
Hi!
I hope you're doing well.
I'm sorry to bother or if this question has already been answered (I did search); but I can't install/run the new version of Fooocus.
I previously used Docker Desktop (WSL2) for running Fooocus, but I couldn't update it and accidentally deleted the whole container - so I decided to run Fooocus on its own, using the run.bat files. (I'd tried it before and worked very well.)

So, I downloaded the  Windows version (from the link on the front page), decompressed it and clicked run.bat (after adding --always-download-new-model).
However, this is the output:
--------------
H:\Fooocus>.\python_embeded\python.exe -s Fooocus\entry_with_update.py --always-download-new-model
Fast-forward merge
Update succeeded.
[System ARGV] ['Fooocus\\entry_with_update.py', '--always-download-new-model']
Python 3.10.9 (tags/v3.10.9:1dd9be6, Dec  6 2022, 20:01:21) [MSC v.1934 64 bit (AMD64)]
Fooocus version: 2.5.2
Error checking version for opencv-contrib-python-headless: No package metadata was found for opencv-contrib-python-headless
Installing requirements
[Cleanup] Attempting to delete content of temp dir C:\Users\IMANMI~1\AppData\Local\Temp\fooocus
[Cleanup] Cleanup successful
Downloading: "https://huggingface.co/mashb1t/misc/resolve/main/xl-to-v1_interposer-v4.0.safetensors" to H:\Fooocus\Fooocus\models\vae_approx\xl-to-v1_interposer-v4.0.safetensors

Traceback (most recent call last):
  File "H:\Fooocus\Fooocus\entry_with_update.py", line 46, in <module>
    from launch import *
  File "H:\Fooocus\Fooocus\launch.py", line 140, in <module>
    config.default_base_model_name, config.checkpoint_downloads = download_models(
  File "H:\Fooocus\Fooocus\launch.py", line 104, in download_models
    load_file_from_url(url=url, model_dir=config.path_vae_approx, file_name=file_name)
  File "H:\Fooocus\Fooocus\modules\model_loader.py", line 26, in load_file_from_url
    from torch.hub import download_url_to_file
  File "H:\Fooocus\python_embeded\lib\site-packages\torch\__init__.py", line 137, in <module>
    raise err
OSError: [WinError 127] The specified procedure could not be found. Error loading "H:\Fooocus\python_embeded\lib\site-packages\torch\lib\c10_cuda.dll" or one of its dependencies.

H:\Fooocus>pause
Press any key to continue . . .
---------------
My system's drivers are up to date.
It's a dual CPU (Xeon) motherboard (56 cores), 128 GB RAM, 2 GPUs (Nvidia RTX 3060 12 GB DualOC).
OS is Windows 11 Enterprise x64, 23H2

Thankful in advance if anyone can please help me to make this work. (Again...)
Sincerely,
I

EDIT:
I tried a second time by installing the requirements manually and it downloaded the model - but fails to start:

H:\Fooocus>dir
 Volume in drive H is Storage 2
 Volume Serial Number is FAF1-3C72

 Directory of H:\Fooocus

2024-07-31  22:52    <DIR>          .
2024-07-31  23:02    <DIR>          Fooocus
2024-07-31  22:53    <DIR>          python_embeded
2024-07-31  23:00                96 run.bat
2023-10-14  03:47                85 run_anime.bat
2023-10-14  03:47                89 run_realistic.bat
               3 File(s)            270 bytes
               3 Dir(s)  5 247 168 348 160 bytes free

H:\Fooocus>cd Fooocus

H:\Fooocus\Fooocus>..\python_embeded\python.exe -m pip install -r .\requirements_versions.txt
Requirement already satisfied: torchsde==0.2.6 in h:\fooocus\python_embeded\lib\site-packages (from -r .\requirements_versions.txt (line 1)) (0.2.6)
Requirement already satisfied: einops==0.8.0 in h:\fooocus\python_embeded\lib\site-packages (from -r .\requirements_versions.txt (line 2)) (0.8.0)
Requirement already satisfied: transformers==4.42.4 in h:\fooocus\python_embeded\lib\site-packages (from -r .\requirements_versions.txt (line 3)) (4.42.4)
Requirement already satisfied: safetensors==0.4.3 in h:\fooocus\python_embeded\lib\site-packages (from -r .\requirements_versions.txt (line 4)) (0.4.3)
Requirement already satisfied: accelerate==0.32.1 in h:\fooocus\python_embeded\lib\site-packages (from -r .\requirements_versions.txt (line 5)) (0.32.1)
Requirement already satisfied: pyyaml==6.0.1 in c:\users\iman mirbioki\appdata\roaming\python\python310\site-packages (from -r .\requirements_versions.txt (line 6)) (6.0.1)
Requirement already satisfied: pillow==10.4.0 in h:\fooocus\python_embeded\lib\site-packages (from -r .\requirements_versions.txt (line 7)) (10.4.0)
Requirement already satisfied: scipy==1.14.0 in h:\fooocus\python_embeded\lib\site-packages (from -r .\requirements_versions.txt (line 8)) (1.14.0)
Requirement already satisfied: tqdm==4.66.4 in h:\fooocus\python_embeded\lib\site-packages (from -r .\requirements_versions.txt (line 9)) (4.66.4)
Requirement already satisfied: psutil==6.0.0 in h:\fooocus\python_embeded\lib\site-packages (from -r .\requirements_versions.txt (line 10)) (6.0.0)
Requirement already satisfied: pytorch_lightning==2.3.3 in h:\fooocus\python_embeded\lib\site-packages (from -r .\requirements_versions.txt (line 11)) (2.3.3)
Requirement already satisfied: omegaconf==2.3.0 in h:\fooocus\python_embeded\lib\site-packages (from -r .\requirements_versions.txt (line 12)) (2.3.0)
Requirement already satisfied: gradio==3.41.2 in h:\fooocus\python_embeded\lib\site-packages (from -r .\requirements_versions.txt (line 13)) (3.41.2)
Requirement already satisfied: pygit2==1.15.1 in h:\fooocus\python_embeded\lib\site-packages (from -r .\requirements_versions.txt (line 14)) (1.15.1)
Requirement already satisfied: opencv-contrib-python-headless==4.10.0.84 in h:\fooocus\python_embeded\lib\site-packages (from -r .\requirements_versions.txt (line 15)) (4.10.0.84)
Requirement already satisfied: httpx==0.27.0 in c:\users\iman mirbioki\appdata\roaming\python\python310\site-packages (from -r .\requirements_versions.txt (line 16)) (0.27.0)
Requirement already satisfied: onnxruntime==1.18.1 in h:\fooocus\python_embeded\lib\site-packages (from -r .\requirements_versions.txt (line 17)) (1.18.1)
Requirement already satisfied: timm==1.0.7 in h:\fooocus\python_embeded\lib\site-packages (from -r .\requirements_versions.txt (line 18)) (1.0.7)
Requirement already satisfied: numpy==1.26.4 in h:\fooocus\python_embeded\lib\site-packages (from -r .\requirements_versions.txt (line 19)) (1.26.4)
Requirement already satisfied: tokenizers==0.19.1 in h:\fooocus\python_embeded\lib\site-packages (from -r .\requirements_versions.txt (line 20)) (0.19.1)
Requirement already satisfied: packaging==24.1 in h:\fooocus\python_embeded\lib\site-packages (from -r .\requirements_versions.txt (line 21)) (24.1)
Requirement already satisfied: rembg==2.0.57 in h:\fooocus\python_embeded\lib\site-packages (from -r .\requirements_versions.txt (line 22)) (2.0.57)
Requirement already satisfied: groundingdino-py==0.4.0 in h:\fooocus\python_embeded\lib\site-packages (from -r .\requirements_versions.txt (line 23)) (0.4.0)
Requirement already satisfied: segment_anything==1.0 in h:\fooocus\python_embeded\lib\site-packages (from -r .\requirements_versions.txt (line 24)) (1.0)
Requirement already satisfied: torch>=1.6.0 in h:\fooocus\python_embeded\lib\site-packages (from torchsde==0.2.6->-r .\requirements_versions.txt (line 1)) (2.1.0+cu121)
Requirement already satisfied: trampoline>=0.1.2 in h:\fooocus\python_embeded\lib\site-packages (from torchsde==0.2.6->-r .\requirements_versions.txt (line 1)) (0.1.2)
Requirement already satisfied: filelock in c:\users\iman mirbioki\appdata\roaming\python\python310\site-packages (from transformers==4.42.4->-r .\requirements_versions.txt (line 3)) (3.13.3)
Requirement already satisfied: huggingface-hub<1.0,>=0.23.2 in h:\fooocus\python_embeded\lib\site-packages (from transformers==4.42.4->-r .\requirements_versions.txt (line 3)) (0.23.5)
Requirement already satisfied: regex!=2019.12.17 in c:\users\iman mirbioki\appdata\roaming\python\python310\site-packages (from transformers==4.42.4->-r .\requirements_versions.txt (line 3)) (2023.12.25)
Requirement already satisfied: requests in c:\users\iman mirbioki\appdata\roaming\python\python310\site-packages (from transformers==4.42.4->-r .\requirements_versions.txt (line 3)) (2.31.0)
Requirement already satisfied: colorama in c:\users\iman mirbioki\appdata\roaming\python\python310\site-packages (from tqdm==4.66.4->-r .\requirements_versions.txt (line 9)) (0.4.6)
Requirement already satisfied: fsspec>=2022.5.0 in c:\users\iman mirbioki\appdata\roaming\python\python310\site-packages (from fsspec[http]>=2022.5.0->pytorch_lightning==2.3.3->-r .\requirements_versions.txt (line 11)) (2024.3.1)
Requirement already satisfied: torchmetrics>=0.7.0 in h:\fooocus\python_embeded\lib\site-packages (from pytorch_lightning==2.3.3->-r .\requirements_versions.txt (line 11)) (1.0.3)
Requirement already satisfied: typing-extensions>=4.4.0 in c:\users\iman mirbioki\appdata\roaming\python\python310\site-packages (from pytorch_lightning==2.3.3->-r .\requirements_versions.txt (line 11)) (4.10.0)
Requirement already satisfied: lightning-utilities>=0.10.0 in h:\fooocus\python_embeded\lib\site-packages (from pytorch_lightning==2.3.3->-r .\requirements_versions.txt (line 11)) (0.11.5)
Requirement already satisfied: antlr4-python3-runtime==4.9.* in h:\fooocus\python_embeded\lib\site-packages (from omegaconf==2.3.0->-r .\requirements_versions.txt (line 12)) (4.9.3)
Requirement already satisfied: aiofiles<24.0,>=22.0 in h:\fooocus\python_embeded\lib\site-packages (from gradio==3.41.2->-r .\requirements_versions.txt (line 13)) (23.2.1)
Requirement already satisfied: altair<6.0,>=4.2.0 in h:\fooocus\python_embeded\lib\site-packages (from gradio==3.41.2->-r .\requirements_versions.txt (line 13)) (5.0.1)
Requirement already satisfied: fastapi in h:\fooocus\python_embeded\lib\site-packages (from gradio==3.41.2->-r .\requirements_versions.txt (line 13)) (0.101.0)
Requirement already satisfied: ffmpy in h:\fooocus\python_embeded\lib\site-packages (from gradio==3.41.2->-r .\requirements_versions.txt (line 13)) (0.3.1)
Requirement already satisfied: gradio-client==0.5.0 in h:\fooocus\python_embeded\lib\site-packages (from gradio==3.41.2->-r .\requirements_versions.txt (line 13)) (0.5.0)
Requirement already satisfied: importlib-resources<7.0,>=1.3 in h:\fooocus\python_embeded\lib\site-packages (from gradio==3.41.2->-r .\requirements_versions.txt (line 13)) (6.4.0)
Requirement already satisfied: jinja2<4.0 in c:\users\iman mirbioki\appdata\roaming\python\python310\site-packages (from gradio==3.41.2->-r .\requirements_versions.txt (line 13)) (3.1.3)
Requirement already satisfied: markupsafe~=2.0 in c:\users\iman mirbioki\appdata\roaming\python\python310\site-packages (from gradio==3.41.2->-r .\requirements_versions.txt (line 13)) (2.1.5)
Requirement already satisfied: matplotlib~=3.0 in h:\fooocus\python_embeded\lib\site-packages (from gradio==3.41.2->-r .\requirements_versions.txt (line 13)) (3.7.2)
Requirement already satisfied: orjson~=3.0 in h:\fooocus\python_embeded\lib\site-packages (from gradio==3.41.2->-r .\requirements_versions.txt (line 13)) (3.9.4)
Requirement already satisfied: pandas<3.0,>=1.0 in h:\fooocus\python_embeded\lib\site-packages (from gradio==3.41.2->-r .\requirements_versions.txt (line 13)) (2.0.3)
Requirement already satisfied: pydantic!=1.8,!=1.8.1,!=2.0.0,!=2.0.1,<3.0.0,>=1.7.4 in c:\users\iman mirbioki\appdata\roaming\python\python310\site-packages (from gradio==3.41.2->-r .\requirements_versions.txt (line 13)) (2.6.4)
Requirement already satisfied: pydub in h:\fooocus\python_embeded\lib\site-packages (from gradio==3.41.2->-r .\requirements_versions.txt (line 13)) (0.25.1)
Requirement already satisfied: python-multipart in h:\fooocus\python_embeded\lib\site-packages (from gradio==3.41.2->-r .\requirements_versions.txt (line 13)) (0.0.6)
Requirement already satisfied: semantic-version~=2.0 in h:\fooocus\python_embeded\lib\site-packages (from gradio==3.41.2->-r .\requirements_versions.txt (line 13)) (2.10.0)
Requirement already satisfied: uvicorn>=0.14.0 in h:\fooocus\python_embeded\lib\site-packages (from gradio==3.41.2->-r .\requirements_versions.txt (line 13)) (0.23.2)
Requirement already satisfied: websockets<12.0,>=10.0 in h:\fooocus\python_embeded\lib\site-packages (from gradio==3.41.2->-r .\requirements_versions.txt (line 13)) (11.0.3)
Requirement already satisfied: cffi>=1.16.0 in c:\users\iman mirbioki\appdata\roaming\python\python310\site-packages (from pygit2==1.15.1->-r .\requirements_versions.txt (line 14)) (1.16.0)
Requirement already satisfied: anyio in c:\users\iman mirbioki\appdata\roaming\python\python310\site-packages (from httpx==0.27.0->-r .\requirements_versions.txt (line 16)) (4.3.0)
Requirement already satisfied: certifi in c:\users\iman mirbioki\appdata\roaming\python\python310\site-packages (from httpx==0.27.0->-r .\requirements_versions.txt (line 16)) (2024.2.2)
Requirement already satisfied: httpcore==1.* in c:\users\iman mirbioki\appdata\roaming\python\python310\site-packages (from httpx==0.27.0->-r .\requirements_versions.txt (line 16)) (1.0.4)
Requirement already satisfied: idna in c:\users\iman mirbioki\appdata\roaming\python\python310\site-packages (from httpx==0.27.0->-r .\requirements_versions.txt (line 16)) (3.6)
Requirement already satisfied: sniffio in c:\users\iman mirbioki\appdata\roaming\python\python310\site-packages (from httpx==0.27.0->-r .\requirements_versions.txt (line 16)) (1.3.1)
Requirement already satisfied: coloredlogs in h:\fooocus\python_embeded\lib\site-packages (from onnxruntime==1.18.1->-r .\requirements_versions.txt (line 17)) (15.0.1)
Requirement already satisfied: flatbuffers in h:\fooocus\python_embeded\lib\site-packages (from onnxruntime==1.18.1->-r .\requirements_versions.txt (line 17)) (23.5.26)
Requirement already satisfied: protobuf in c:\users\iman mirbioki\appdata\roaming\python\python310\site-packages (from onnxruntime==1.18.1->-r .\requirements_versions.txt (line 17)) (3.19.6)
Requirement already satisfied: sympy in c:\users\iman mirbioki\appdata\roaming\python\python310\site-packages (from onnxruntime==1.18.1->-r .\requirements_versions.txt (line 17)) (1.12)
Requirement already satisfied: torchvision in h:\fooocus\python_embeded\lib\site-packages (from timm==1.0.7->-r .\requirements_versions.txt (line 18)) (0.19.0)
Requirement already satisfied: jsonschema in h:\fooocus\python_embeded\lib\site-packages (from rembg==2.0.57->-r .\requirements_versions.txt (line 22)) (4.19.0)
Requirement already satisfied: opencv-python-headless in h:\fooocus\python_embeded\lib\site-packages (from rembg==2.0.57->-r .\requirements_versions.txt (line 22)) (4.10.0.84)
Requirement already satisfied: pooch in h:\fooocus\python_embeded\lib\site-packages (from rembg==2.0.57->-r .\requirements_versions.txt (line 22)) (1.8.2)
Requirement already satisfied: pymatting in h:\fooocus\python_embeded\lib\site-packages (from rembg==2.0.57->-r .\requirements_versions.txt (line 22)) (1.1.12)
Requirement already satisfied: scikit-image in h:\fooocus\python_embeded\lib\site-packages (from rembg==2.0.57->-r .\requirements_versions.txt (line 22)) (0.24.0)
Requirement already satisfied: addict in h:\fooocus\python_embeded\lib\site-packages (from groundingdino-py==0.4.0->-r .\requirements_versions.txt (line 23)) (2.4.0)
Requirement already satisfied: yapf in h:\fooocus\python_embeded\lib\site-packages (from groundingdino-py==0.4.0->-r .\requirements_versions.txt (line 23)) (0.40.2)
Requirement already satisfied: opencv-python in c:\users\iman mirbioki\appdata\roaming\python\python310\site-packages (from groundingdino-py==0.4.0->-r .\requirements_versions.txt (line 23)) (4.9.0.80)
Requirement already satisfied: supervision==0.6.0 in h:\fooocus\python_embeded\lib\site-packages (from groundingdino-py==0.4.0->-r .\requirements_versions.txt (line 23)) (0.6.0)
Requirement already satisfied: pycocotools in h:\fooocus\python_embeded\lib\site-packages (from groundingdino-py==0.4.0->-r .\requirements_versions.txt (line 23)) (2.0.8)
Requirement already satisfied: h11<0.15,>=0.13 in c:\users\iman mirbioki\appdata\roaming\python\python310\site-packages (from httpcore==1.*->httpx==0.27.0->-r .\requirements_versions.txt (line 16)) (0.14.0)
Requirement already satisfied: toolz in h:\fooocus\python_embeded\lib\site-packages (from altair<6.0,>=4.2.0->gradio==3.41.2->-r .\requirements_versions.txt (line 13)) (0.12.0)
Requirement already satisfied: pycparser in c:\users\iman mirbioki\appdata\roaming\python\python310\site-packages (from cffi>=1.16.0->pygit2==1.15.1->-r .\requirements_versions.txt (line 14)) (2.22)
Requirement already satisfied: aiohttp!=4.0.0a0,!=4.0.0a1 in h:\fooocus\python_embeded\lib\site-packages (from fsspec[http]>=2022.5.0->pytorch_lightning==2.3.3->-r .\requirements_versions.txt (line 11)) (3.8.4)
Requirement already satisfied: attrs>=22.2.0 in h:\fooocus\python_embeded\lib\site-packages (from jsonschema->rembg==2.0.57->-r .\requirements_versions.txt (line 22)) (23.1.0)
Requirement already satisfied: jsonschema-specifications>=2023.03.6 in h:\fooocus\python_embeded\lib\site-packages (from jsonschema->rembg==2.0.57->-r .\requirements_versions.txt (line 22)) (2023.7.1)
Requirement already satisfied: referencing>=0.28.4 in h:\fooocus\python_embeded\lib\site-packages (from jsonschema->rembg==2.0.57->-r .\requirements_versions.txt (line 22)) (0.30.2)
Requirement already satisfied: rpds-py>=0.7.1 in h:\fooocus\python_embeded\lib\site-packages (from jsonschema->rembg==2.0.57->-r .\requirements_versions.txt (line 22)) (0.9.2)
Requirement already satisfied: setuptools in h:\fooocus\python_embeded\lib\site-packages (from lightning-utilities>=0.10.0->pytorch_lightning==2.3.3->-r .\requirements_versions.txt (line 11)) (68.0.0)
Requirement already satisfied: contourpy>=1.0.1 in h:\fooocus\python_embeded\lib\site-packages (from matplotlib~=3.0->gradio==3.41.2->-r .\requirements_versions.txt (line 13)) (1.1.0)
Requirement already satisfied: cycler>=0.10 in h:\fooocus\python_embeded\lib\site-packages (from matplotlib~=3.0->gradio==3.41.2->-r .\requirements_versions.txt (line 13)) (0.11.0)
Requirement already satisfied: fonttools>=4.22.0 in h:\fooocus\python_embeded\lib\site-packages (from matplotlib~=3.0->gradio==3.41.2->-r .\requirements_versions.txt (line 13)) (4.42.0)
Requirement already satisfied: kiwisolver>=1.0.1 in h:\fooocus\python_embeded\lib\site-packages (from matplotlib~=3.0->gradio==3.41.2->-r .\requirements_versions.txt (line 13)) (1.4.4)
Requirement already satisfied: pyparsing<3.1,>=2.3.1 in h:\fooocus\python_embeded\lib\site-packages (from matplotlib~=3.0->gradio==3.41.2->-r .\requirements_versions.txt (line 13)) (3.0.9)
Requirement already satisfied: python-dateutil>=2.7 in h:\fooocus\python_embeded\lib\site-packages (from matplotlib~=3.0->gradio==3.41.2->-r .\requirements_versions.txt (line 13)) (2.8.2)
Requirement already satisfied: pytz>=2020.1 in h:\fooocus\python_embeded\lib\site-packages (from pandas<3.0,>=1.0->gradio==3.41.2->-r .\requirements_versions.txt (line 13)) (2023.3)
Requirement already satisfied: tzdata>=2022.1 in h:\fooocus\python_embeded\lib\site-packages (from pandas<3.0,>=1.0->gradio==3.41.2->-r .\requirements_versions.txt (line 13)) (2023.3)
Requirement already satisfied: annotated-types>=0.4.0 in c:\users\iman mirbioki\appdata\roaming\python\python310\site-packages (from pydantic!=1.8,!=1.8.1,!=2.0.0,!=2.0.1,<3.0.0,>=1.7.4->gradio==3.41.2->-r .\requirements_versions.txt (line 13)) (0.6.0)
Requirement already satisfied: pydantic-core==2.16.3 in c:\users\iman mirbioki\appdata\roaming\python\python310\site-packages (from pydantic!=1.8,!=1.8.1,!=2.0.0,!=2.0.1,<3.0.0,>=1.7.4->gradio==3.41.2->-r .\requirements_versions.txt (line 13)) (2.16.3)
Requirement already satisfied: charset-normalizer<4,>=2 in c:\users\iman mirbioki\appdata\roaming\python\python310\site-packages (from requests->transformers==4.42.4->-r .\requirements_versions.txt (line 3)) (3.3.2)
Requirement already satisfied: urllib3<3,>=1.21.1 in c:\users\iman mirbioki\appdata\roaming\python\python310\site-packages (from requests->transformers==4.42.4->-r .\requirements_versions.txt (line 3)) (2.2.1)
Requirement already satisfied: networkx in c:\users\iman mirbioki\appdata\roaming\python\python310\site-packages (from torch>=1.6.0->torchsde==0.2.6->-r .\requirements_versions.txt (line 1)) (3.2.1)
Requirement already satisfied: click>=7.0 in c:\users\iman mirbioki\appdata\roaming\python\python310\site-packages (from uvicorn>=0.14.0->gradio==3.41.2->-r .\requirements_versions.txt (line 13)) (8.1.7)
Requirement already satisfied: exceptiongroup>=1.0.2 in c:\users\iman mirbioki\appdata\roaming\python\python310\site-packages (from anyio->httpx==0.27.0->-r .\requirements_versions.txt (line 16)) (1.2.0)
Requirement already satisfied: humanfriendly>=9.1 in h:\fooocus\python_embeded\lib\site-packages (from coloredlogs->onnxruntime==1.18.1->-r .\requirements_versions.txt (line 17)) (10.0)
Requirement already satisfied: starlette<0.28.0,>=0.27.0 in h:\fooocus\python_embeded\lib\site-packages (from fastapi->gradio==3.41.2->-r .\requirements_versions.txt (line 13)) (0.27.0)
Requirement already satisfied: platformdirs>=2.5.0 in c:\users\iman mirbioki\appdata\roaming\python\python310\site-packages (from pooch->rembg==2.0.57->-r .\requirements_versions.txt (line 22)) (4.2.2)
Requirement already satisfied: numba!=0.49.0 in h:\fooocus\python_embeded\lib\site-packages (from pymatting->rembg==2.0.57->-r .\requirements_versions.txt (line 22)) (0.58.1)
Requirement already satisfied: imageio>=2.33 in h:\fooocus\python_embeded\lib\site-packages (from scikit-image->rembg==2.0.57->-r .\requirements_versions.txt (line 22)) (2.34.2)
Requirement already satisfied: tifffile>=2022.8.12 in h:\fooocus\python_embeded\lib\site-packages (from scikit-image->rembg==2.0.57->-r .\requirements_versions.txt (line 22)) (2024.7.2)
Requirement already satisfied: lazy-loader>=0.4 in h:\fooocus\python_embeded\lib\site-packages (from scikit-image->rembg==2.0.57->-r .\requirements_versions.txt (line 22)) (0.4)
Requirement already satisfied: mpmath>=0.19 in c:\users\iman mirbioki\appdata\roaming\python\python310\site-packages (from sympy->onnxruntime==1.18.1->-r .\requirements_versions.txt (line 17)) (1.3.0)
Collecting torch>=1.6.0 (from torchsde==0.2.6->-r .\requirements_versions.txt (line 1))
  Using cached torch-2.4.0-cp310-cp310-win_amd64.whl.metadata (27 kB)
Requirement already satisfied: importlib-metadata>=6.6.0 in h:\fooocus\python_embeded\lib\site-packages (from yapf->groundingdino-py==0.4.0->-r .\requirements_versions.txt (line 23)) (8.0.0)
Requirement already satisfied: tomli>=2.0.1 in h:\fooocus\python_embeded\lib\site-packages (from yapf->groundingdino-py==0.4.0->-r .\requirements_versions.txt (line 23)) (2.0.1)
Requirement already satisfied: multidict<7.0,>=4.5 in h:\fooocus\python_embeded\lib\site-packages (from aiohttp!=4.0.0a0,!=4.0.0a1->fsspec[http]>=2022.5.0->pytorch_lightning==2.3.3->-r .\requirements_versions.txt (line 11)) (6.0.4)
Requirement already satisfied: async-timeout<5.0,>=4.0.0a3 in h:\fooocus\python_embeded\lib\site-packages (from aiohttp!=4.0.0a0,!=4.0.0a1->fsspec[http]>=2022.5.0->pytorch_lightning==2.3.3->-r .\requirements_versions.txt (line 11)) (4.0.2)
Requirement already satisfied: yarl<2.0,>=1.0 in h:\fooocus\python_embeded\lib\site-packages (from aiohttp!=4.0.0a0,!=4.0.0a1->fsspec[http]>=2022.5.0->pytorch_lightning==2.3.3->-r .\requirements_versions.txt (line 11)) (1.9.2)
Requirement already satisfied: frozenlist>=1.1.1 in h:\fooocus\python_embeded\lib\site-packages (from aiohttp!=4.0.0a0,!=4.0.0a1->fsspec[http]>=2022.5.0->pytorch_lightning==2.3.3->-r .\requirements_versions.txt (line 11)) (1.3.3)
Requirement already satisfied: aiosignal>=1.1.2 in h:\fooocus\python_embeded\lib\site-packages (from aiohttp!=4.0.0a0,!=4.0.0a1->fsspec[http]>=2022.5.0->pytorch_lightning==2.3.3->-r .\requirements_versions.txt (line 11)) (1.3.1)
Requirement already satisfied: pyreadline3 in h:\fooocus\python_embeded\lib\site-packages (from humanfriendly>=9.1->coloredlogs->onnxruntime==1.18.1->-r .\requirements_versions.txt (line 17)) (3.4.1)
Requirement already satisfied: zipp>=0.5 in h:\fooocus\python_embeded\lib\site-packages (from importlib-metadata>=6.6.0->yapf->groundingdino-py==0.4.0->-r .\requirements_versions.txt (line 23)) (3.19.2)
Requirement already satisfied: llvmlite<0.42,>=0.41.0dev0 in h:\fooocus\python_embeded\lib\site-packages (from numba!=0.49.0->pymatting->rembg==2.0.57->-r .\requirements_versions.txt (line 22)) (0.41.1)
Requirement already satisfied: six>=1.5 in h:\fooocus\python_embeded\lib\site-packages (from python-dateutil>=2.7->matplotlib~=3.0->gradio==3.41.2->-r .\requirements_versions.txt (line 13)) (1.16.0)
Using cached torch-2.4.0-cp310-cp310-win_amd64.whl (197.9 MB)
Installing collected packages: torch
  Attempting uninstall: torch
    Found existing installation: torch 2.1.0+cu121
    Uninstalling torch-2.1.0+cu121:
      Successfully uninstalled torch-2.1.0+cu121
  WARNING: The scripts convert-caffe2-to-onnx.exe, convert-onnx-to-caffe2.exe and torchrun.exe are installed in 'H:\Fooocus\python_embeded\Scripts' which is not on PATH.
  Consider adding this directory to PATH or, if you prefer to suppress this warning, use --no-warn-script-location.
Successfully installed torch-2.4.0

[notice] A new release of pip is available: 24.1 -> 24.2
[notice] To update, run: H:\Fooocus\python_embeded\python.exe -m pip install --upgrade pip

H:\Fooocus\Fooocus>cd ..

H:\Fooocus>run.bat

H:\Fooocus>.\python_embeded\python.exe -s Fooocus\entry_with_update.py --always-download-new-model
Already up-to-date
Update succeeded.
[System ARGV] ['Fooocus\\entry_with_update.py', '--always-download-new-model']
Python 3.10.9 (tags/v3.10.9:1dd9be6, Dec  6 2022, 20:01:21) [MSC v.1934 64 bit (AMD64)]
Fooocus version: 2.5.2
[Cleanup] Attempting to delete content of temp dir C:\Users\IMANMI~1\AppData\Local\Temp\fooocus
[Cleanup] Cleanup successful
Downloading: "https://huggingface.co/mashb1t/misc/resolve/main/xl-to-v1_interposer-v4.0.safetensors" to H:\Fooocus\Fooocus\models\vae_approx\xl-to-v1_interposer-v4.0.safetensors

100%|█████████████████████████████████████████████████████████████████████████████| 5.40M/5.40M [00:00<00:00, 42.3MB/s]
Downloading: "https://huggingface.co/lllyasviel/fav_models/resolve/main/fav/juggernautXL_v8Rundiffusion.safetensors" to H:\Fooocus\Fooocus\models\checkpoints\juggernautXL_v8Rundiffusion.safetensors

100%|█████████████████████████████████████████████████████████████████████████████| 6.62G/6.62G [01:27<00:00, 81.6MB/s]
Traceback (most recent call last):
  File "H:\Fooocus\Fooocus\entry_with_update.py", line 46, in <module>
    from launch import *
  File "H:\Fooocus\Fooocus\launch.py", line 147, in <module>
    from webui import *
  File "H:\Fooocus\Fooocus\webui.py", line 10, in <module>
    import modules.async_worker as worker
  File "H:\Fooocus\Fooocus\modules\async_worker.py", line 3, in <module>
    from extras.inpaint_mask import generate_mask_from_image, SAMOptions
  File "H:\Fooocus\Fooocus\extras\inpaint_mask.py", line 6, in <module>
    from extras.GroundingDINO.util.inference import default_groundingdino
  File "H:\Fooocus\Fooocus\extras\GroundingDINO\util\inference.py", line 3, in <module>
    import ldm_patched.modules.model_management as model_management
  File "H:\Fooocus\Fooocus\ldm_patched\modules\model_management.py", line 121, in <module>
    total_vram = get_total_memory(get_torch_device()) / (1024 * 1024)
  File "H:\Fooocus\Fooocus\ldm_patched\modules\model_management.py", line 90, in get_torch_device
    return torch.device(torch.cuda.current_device())
  File "H:\Fooocus\python_embeded\lib\site-packages\torch\cuda\__init__.py", line 878, in current_device
    _lazy_init()
  File "H:\Fooocus\python_embeded\lib\site-packages\torch\cuda\__init__.py", line 305, in _lazy_init
    raise AssertionError("Torch not compiled with CUDA enabled")
AssertionError: Torch not compiled with CUDA enabled

H:\Fooocus>pause
Press any key to continue . . .

**Comment by mashb1t** [comment, marked as answer]
Luckily there is a relatively easy solution for this.
Please modify https://github.com/lllyasviel/Fooocus/blob/main/launch.py#L25 to True and start Fooocus again, which reinstalls torch with compiled cuda.
After that, change the line to False again and you're good to go!

**Comment by catclaw** [comment]
Hi!

Thank you for taking the time to answer my question.
However, unfortunately, the solution did not work.
I get this error message instead:

H:\Fooocus>.\python_embeded\python.exe -s Fooocus\entry_with_update.py
Update failed.
No module named 'pygit2'
Update succeeded.
[System ARGV] ['.\\Fooocus\\entry_with_update.py']
Traceback (most recent call last):
  File "H:\Fooocus\Fooocus\entry_with_update.py", line 46, in <module>
    from launch import *
  File "H:\Fooocus\Fooocus\launch.py", line 22, in <module>
    from modules.launch_util import is_installed, run, python, run_pip, requirements_met, delete_folder_content
  File "H:\Fooocus\Fooocus\modules\launch_util.py", line 10, in <module>
    import packaging.version
ModuleNotFoundError: No module named 'packaging'

H:\Fooocus>pause
Press any key to continue . . .

I managed to install the latest version with Docker Desktop earlier this morning.
The only problem is that the Docker installation is so much slower.

**Reply by mashb1t** [reply]
You may use this command to set up the packages:
`H:\Fooocus\python_embeded\python.exe -m pip install -r H:\Fooocus\Fooocus\requirements_versions.txt`

**Reply by catclaw** [reply]
Thank you!
Yes - I tried that too...
Didn't work?
However, I continue to use Fooocus with Docker Desktop till I can find a solution.
I just got my hands on a PNY Nvidia Tesla A100 80GB - WOOOOOOW!!! This is sick!
My only problem is that Docker only reserves 64 GB RAM out of 128 GB, and 28 cores out of 56 - and I can't change it easily in settings, as it's configured for WSL2. I have to change to global WSL configuration... Frustrating!

Thank you everyone for trying to help me!
All love ❤️🙏
